Adding error subtype for malformed media argument.

develric opened the pull request:
Fixes #13353 . This has a companion [in this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2005).

This PR adds a `MediaErrorSubType` persisted in the `MediaUploadModel` that can be used to better discriminate in the client app an error subcategory and decide to present a localisable and hopefully clearer message to the user.

This PR takes care of the user facing message in the Editor presented in the snackbar and in the dialog that appears when the user taps on a media that had an issue uploading in the post. The overlay message is not covered in this PR and there is a dedicated issue (#13815).

### To test
- Create a new post
- Add an image block and add an image from device
- Navigate to select from Google Photos and select a webp image. Notice you get the expected message in the snackbar and in the Retry dialog

| ![image](https://user-images.githubusercontent.com/47797566/118561954-2466cd00-b76c-11eb-98bb-dcc03aadc9f8.png) | ![image](https://user-images.githubusercontent.com/47797566/118562035-45c7b900-b76c-11eb-866a-032e8d3c1cbd.png) |
|---|---|

- Using the debug you can force other type of errors for `MalformedMediaArgSubType`
- Smoke test both media upload from the Editor and from the Media Library
- Smoke test opening drafts or published posts and editing them
- Smoke test page creation/editing
- Test the db upgrade script: fresh install should create a `MediaUploadModel.java` table containing `ERROR_SUB_TYPE` column. Also upgrading from a previous version of the app you should see that same column is added with NULL values.

## Regression Notes
1. Potential unintended areas of impact
Media upload and Post/Page editing

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Added a bit of unit testing in the FluxC companion PR and manual testing

3. What automated tests I added (or what prevented me from doing so)
Unit tests in FluxC PR.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

develric committed:
Adding relevant strings resources.

develric committed:
Extending error subtype usage to Utils.

develric committed:
Fixing lint.

develric committed:
Setting fluxc hash.

develric committed:
Merge branch 'develop' into issue/13353-improve-mime-type-error-message

develric committed:
Updating fluxc hash.

planarvoid commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -1803,6 +1803,11 @@
     <string name="error_media_quota_exceeded">Storage quota exceeded</string>
     <string name="error_media_quota_exceeded_toast">Cannot upload file.\nStorage quota was exceeded.</string>
     <string name="error_media_not_enough_storage">Not enough available site storage</string>
+    <string name="error_media_unexpected_null_value">Unexpected null value for Media</string>
```
I'd rephrase this to make it more user friendly. What about "Media was empty"?

develric committed:
Updating fluxc hash.

develric committed:
Merge branch 'develop' into issue/13353-improve-mime-type-error-message

# Conflicts:
#	build.gradle

develric committed:
Changing copy as per feedback.

develric commented:
Hey @planarvoid, followed up here and updated the fluxc hash to the new changes. Please note I added this test step 

>Test the db upgrade script: fresh install should create a MediaUploadModel.java table containing ERROR_SUB_TYPE column. Also upgrading from a previous version of the app you should see that same column is added with NULL values.

to account for the db update script. Should be ready for your pass, thanks 😊

develric committed:
Updating fluxc hash.

develric committed:
Updating fluxc hash.

planarvoid reviewed (approved):
It's looking good and working well 👍 , thanks for the changes!

develric committed:
Updating fluxc tag.

develric committed:
Updating fluxc to new tag to fix ci failure.

develric commented:
Just noting here for tracking before to merge this PR that I updated the FluxC tag to include [this fix](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2009) that deals with the CI build issue that was occurring.
